api/src/service/domain/workflow/workflowitem_create.ts

```diff
@@ -183,23 +188,29 @@ export async function createWorkflowitem(
       // preparation for workflowitem_created event
       for (const doc of reqData.documents || []) {
         doc.id = generateUniqueDocId(existingDocuments);
-        const hashedDocumentResult = await hashDocument(doc);
-        if (Result.isErr(hashedDocumentResult)) {
-          return new VError(hashedDocumentResult, `cannot hash document ${doc.id} `);
+        if ("base64" in doc) {
+          const hashedDocumentResult = await hashDocument(doc);
+          if (Result.isErr(hashedDocumentResult)) {
+            return new VError(hashedDocumentResult, `cannot hash document ${doc.id} `);
+          }
+          documents.push(hashedDocumentResult);
+        } else {
+          documents.push(doc);
         }
-        documents.push(hashedDocumentResult);
       }
       // upload documents to storage service
       // generate document events (document_uploaded, secret_published)
       const documentUploadedEventsResults: Result.Type<BusinessEvent[]>[] = await Promise.all(
-        reqData.documents.map(async (document) => {
-          logger.trace({ document }, "Trying to upload document to storage service");
-          return repository.uploadDocumentToStorageService(
-            document.fileName || "",
-            document.base64,
-            document.id,
-          );
-        }),
+        reqData.documents
+          .filter((document) => "base64" in document)
+          .map(async (document) => {
+            logger.trace({ document }, "Trying to upload document to storage service");
+            return repository.uploadDocumentToStorageService(
+              document.fileName || "",
+              document.base64,
+              document.id,
+            );
+          }),
```

Comment:
Not directly associated with the PR but, use `Promise.allSettled() ` instead of `Promise.all()` and check the result / handle error.